activemodel/lib/active_model/type/helpers/time_value.rb

```diff
@@ -65,7 +65,7 @@ def new_time(year, mon, mday, hour, min, sec, microsec, offset = nil)
             \A
             (\d{4})-(\d\d)-(\d\d)(?:T|\s)            # 2020-06-20T
             (\d\d):(\d\d):(\d\d)(?:\.(\d{1,6})\d*)?  # 10:20:30.123456
-            (?:(Z(?=\z)|[+-]\d\d)(?::?(\d\d))?)?     # +09:00
+            (?:(Z|[+-]\d\d)(?::?(\d\d))?)?           # +09:00
```

Comment:
As the author of #39676, I had intended to limit the `$8` capture only `"Z"` or `"\d\d"`.

https://github.com/rails/rails/blob/db4ed19d8c6ce597e879ab729fecae82f6d76a9e/activemodel/lib/active_model/type/helpers/time_value.rb#L81-L83

Removing the `(?=\z)` part will change the parsing behavior not to fallback `fast_string_to_time` to `fallback_string_to_time`.

Reply:
You mean, in case of an invalid input that contains both `Z` and offset? That makes sense. Thank you for the explanation!